js/Worksheet.js

```diff
@@ -625,7 +729,7 @@ class Worksheet extends HTMLElement {
             const sourceName = event.dataTransfer.getData("name");
             this.addSource(sourceId, sourceName);
             // now tell the source to add me as a target
-            // TODO: maybe all of this source/target adding/removing should be
+            // TODO: maybe all of this source/target adding/removing should
```

Comment:
If we create custom elements that each represent a "connection" between two sheets, we can handle this in the `connectedCallback`, `disconnectedCallback`, and `attributeChanged` handlers of those elements -- ie, lookup the elements corresponding to a source and destination and bind events on them as needed.

Reply:
yeah we need to think through the design of what a connection looks like:

* is it source -> target (and the mapper is implied, doesn't even need to be a sheet since it's really the instruction/callstack which is important)
* do we allow multiple sources, and potentially targets or do we force this sort of functionality by coupling multiple source -> temp-target compositions. For example, (sourceA, sourceB) -> target == (sourceB -> (sourceA -> temp-target)). Clearly a nice algebra there to be had but might not be convenient to write... 

i think this is the next step, and that a connection should be it's own object/custom el which is responsible for 
* lookup argument sheets (sources, targets etc)
* handle opening an editor for the callstack, whether this be another sheet or whatever else
* deal with errors
* handle translating keyboard and mouse events into instructions

maybe we can spec this out today/tomorrow and i can write it next week